[TH2-4262] added enabledExternalQueueRouting property (#49)
th2-net/th2-codec: README.md

> **NOTE**: codec can replace raw message with a parsed message followed by several raw messages
> (e.g. when a codec decodes only a transport layer it can produce a parsed message for the transport layer and several raw messages for its payload)

## External queue routing logic

> Codec can send batch to external queue instead of configured by infra schema. The next condition must be met:
> * enabledExternalQueueRouting must be true
> * MessageGroupBatch.metadata.externalUserQueue of income batch must be not blank
> * Processed batch must be transformed completely. For example: 
>   * encoder considers batch as transformed completed when all messages are converted to raw format.
>   * decoder considers batch as transformed completed when all messages are converted to parsed format.
>   * batch isn't considered as transformed if it contains raw and parsed messages.

Reviewer: ```suggestion
Codec can send a batch to an external queue instead of the one configured by infra schema. The following conditions must be met:
* `enabledExternalQueueRouting` setting must be set to `true`
* `MessageGroupBatch.metadata.externalUserQueue` of an incoming batch must be not blank
* batch must be processed completely e.g.: 
  * encoded batch is completely processed when it has no parsed messages
  * decoded batch is completely processed when it has no raw messages
```

## Codec settings

Common codec settings:

> * enabledExternalQueueRouting - option to enable/disable external queue routing logic. Default value is false.

Reviewer: ```suggestion
* enableExternalQueueRouting - enables external queue routing logic. Default value is `false`.
```

  name: codec
spec:
  custom-config:

>     enabledExternalQueueRouting: false

Reviewer: ```suggestion
    enableExternalQueueRouting: false
```

## Changelog

### v4.7.4+

> #### Changed:
> 
> * Implemented logic to send completely transformed batch to external queue instead of configured by schema.

Reviewer: ```suggestion
#### Added:

* Logic to send completely transformed batch to an external queue instead of a schema-configured one.
```
